### PR TITLE
Create exhaust.lua

### PR DESCRIPTION
**Pull request by S-S-X**
Written directly with github editor so not tested at all

### CODE REVIEW

**Comment by BuckarooBanzay** [comment]
pulled this on the test-server but can't really see any difference to before :thinking:

**Comment by S-S-X** [comment]
> pulled this on the test-server but can't really see any difference to before 🤔

Just tested there and yes seems like it does not work... have to check why, I think I checked that callback function should be always called but there could be just some stupid mistake somewhere.

File: exhaust.lua [hunk 1]
```diff
@@ -0,0 +1,16 @@
+local timestamp = {}
+stamina.register_on_exhaust_player(function(player)
+	if not player then return end
+	local now = minetest.get_us_time()
+	if timestamp[player:get_player_name()] then
+		if now - timestamp[player:get_player_name()] > 999999 then
+			local exhaustion = stamina.get_exhaustion(player) or 0
+			if exhaustion >= stamina.settings.exhaust_lvl then
```
Comment (S-S-X) [inline exhaust.lua]: Seems like this check always fails, not exactly sure what exhaustion actually represesnts. Testing on local server stamina.settings.exhaust_lvl is 160 and stamina.get_exhaustion(player) returns something between 0 and 140.

Comment (S-S-X) [reply exhaust.lua]: Exhaustion is incremented by some value depending on what player is doing and gets reset when eating (and by some other things I'm not exactly sure about).
Seems like should use `change` here instead.

**Comment by S-S-X** [comment]
Now actually tested bit and seems to work well.

**Review by BuckarooBanzay — APPROVED** [review]

**Pull request merged**